portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,29 +29,17 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery res = datastore.prepare(query);
-
-    List<Entity> comments = new ArrayList<>();
-    for(Entity entity : res.asIterable()) {
-        String content = (String) entity.getProperty("content");
-        long timestamp = (long) entity.getProperty("timestamp");
-
-        Entity comment = new Entity(content, timestamp);
-        comments.add(comment);
-    }
     Gson gson = new Gson();
 
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(comments));
+    response.getWriter().println(gson.toJson(getComments(getRequestNum(request))));
```

Comment:
I think this would be more readable on separate lines. It would be no more or less efficient, so the only thing left to optimize for is human readability.